libs/isograph-react/src/core/areEqualWithDeepComparison.ts

```diff
@@ -42,29 +38,34 @@ export function areEqualWithDeepComparison(
   }
 }
 
-export function areEqualArraysWithDeepComparison(
+export function mergeArraysUsingReaderAst(
```

Comment:
It's probably not worth it to recycle all elements in an array, since it would be O(n^2).

But there's a potential to recycle `Nodes` using `id` in O(n).

Reply:
I'm fine with just recycling if the new items are appended (i.e. they're only recycled if their order is the same).

But if you want to implement an O(1) solution, that would be awesome, and will probably help make the situation where users prepend more performant. Let's do it as a separate PR.

I'm not sure there's an obvious circumstance in which someone puts a specific item from an array into a dependency array, though. You would probably do something like `users.find(...)` or `users[0]` or something. I suppose that's not impossible, but doesn't seem like a common pattern.